### ISSUE TITLE
occasionally cannot stop docker containers on windows machines

### ISSUE THREAD

**Issue by krishna81m** [docker/docker#19578]
After this issue https://github.com/docker/machine/issues/2707, now I see docker stop container hanging forever.

```
MINGW64 ~
$ docker rm 34d4b228cd55
Error response from daemon: Conflict, You cannot remove a running container. Stop the container before attempting removal or use -f
Error: failed to remove containers: [34d4b228cd55]

MINGW64 ~
$ docker stop 34d4b228cd55
```

hangs forever, cannot remove it forcefully, only option is restart my laptop....

```
default docker-machine logs

autologin        boot2docker.log  docker.log       parallels.log    udhcp.log        userdata.log     wtmp
docker@default:~$ tail -f /var/log/*.log
==> /var/log/boot2docker.log <==
       valid_lft forever preferred_lft forever
4: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:19:6b:bf brd ff:ff:ff:ff:ff:ff
    inet 192.168.99.100/24 brd 192.168.99.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe19:6bbf/64 scope link
       valid_lft forever preferred_lft forever
-------------------
Need TLS certs for default,127.0.0.1,10.0.2.15,192.168.99.100
-------------------

==> /var/log/docker.log <==
time="2016-01-22T08:40:57.092680613Z" level=debug msg="Calling GET /v1.21/containers/json"
time="2016-01-22T08:40:57.092721102Z" level=info msg="GET /v1.21/containers/json"
time="2016-01-22T08:40:57.092736471Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.0, server: 1.9.1)"

==> /var/log/udhcp.log <==
Fri Jan 22 08:41:00 UTC 2016 ifconfig eth1 192.168.99.100  netmask 255.255.255.0

==> /var/log/docker.log <==
time="2016-01-22T08:41:14.747238390Z" level=debug msg="Calling DELETE /v1.21/containers/34d4b228cd55"
time="2016-01-22T08:41:14.747277670Z" level=info msg="DELETE /v1.21/containers/34d4b228cd55"
time="2016-01-22T08:41:14.747292664Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.0, server: 1.9.1)"
time="2016-01-22T08:41:14.747404328Z" level=error msg="Handler for DELETE /v1.21/containers/34d4b228cd55 returned error: Conflict, You cannot remove a running container. Stop the container before attempting removal or use -f"
time="2016-01-22T08:41:14.747419999Z" level=error msg="HTTP Error" err="Conflict, You cannot remove a running container. Stop the container before attempting removal or use -f" statusCode=409
time="2016-01-22T08:41:25.470196710Z" level=debug msg="Calling POST /v1.21/containers/34d4b228cd55/stop"
time="2016-01-22T08:41:25.470234491Z" level=info msg="POST /v1.21/containers/34d4b228cd55/stop?t=10"
time="2016-01-22T08:41:25.470249705Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.0, server: 1.9.1)"
time="2016-01-22T08:41:25.470398875Z" level=debug msg="Sending 15 to 34d4b228cd55046856411b63fd52c815f794b382f3fe577eba4cb9c150b6b618"
time="2016-01-22T08:41:35.470812965Z" level=info msg="Container 34d4b228cd55046856411b63fd52c815f794b382f3fe577eba4cb9c150b6b618 failed to exit within 10 seconds of SIGTERM - using the force"
time="2016-01-22T08:41:35.470856447Z" level=debug msg="Sending 9 to 34d4b228cd55046856411b63fd52c815f794b382f3fe577eba4cb9c150b6b618"
time="2016-01-22T08:41:45.471442259Z" level=info msg="Container 34d4b228cd55 failed to exit within 10 seconds of kill - trying direct SIGKILL"

==> /var/log/parallels.log <==

==> /var/log/udhcp.log <==
Fri Jan 22 08:00:57 UTC 2016 ifconfig eth0 10.0.2.15  netmask 255.255.255.0
Fri Jan 22 08:00:57 UTC 2016 deleting routers
Fri Jan 22 08:00:57 UTC 2016 route add default gw 10.0.2.2 dev eth0 metric 0
Fri Jan 22 08:00:57 UTC 2016 reset /etc/resolv.conf
Fri Jan 22 08:00:57 UTC 2016 adding dns 192.168.1.1
Fri Jan 22 08:00:58 UTC 2016 ifconfig eth1 192.168.99.100  netmask 255.255.255.0
Fri Jan 22 08:11:00 UTC 2016 ifconfig eth1 192.168.99.100  netmask 255.255.255.0
Fri Jan 22 08:21:00 UTC 2016 ifconfig eth1 192.168.99.100  netmask 255.255.255.0
Fri Jan 22 08:31:00 UTC 2016 ifconfig eth1 192.168.99.100  netmask 255.255.255.0

==> /var/log/userdata.log <==
tar: A lone zero block at 8

```

Environment: Windows 7 laptop
```
$ docker version
Client:
Version: 1.9.0
API version: 1.21
Go version: go1.4.3
Git commit: 76d6bc9
Built: Tue Nov 3 19:20:09 UTC 2015
OS/Arch: windows/amd64

Server:
Version: 1.9.1
API version: 1.21
Go version: go1.4.3
Git commit: a34a1d5
Built: Fri Nov 20 17:56:04 UTC 2015
OS/Arch: linux/amd64

$ docker info
Containers: 32
Images: 279
Server Version: 1.9.1
Storage Driver: aufs
Root Dir: /mnt/sda1/var/lib/docker/aufs
Backing Filesystem: extfs
Dirs: 343
Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.13-boot2docker
Operating System: Boot2Docker 1.9.1 (TCL 6.4.1); master : cef800b - Fri Nov 20 19:33:59 UTC 2015
CPUs: 1
Total Memory: 1.956 GiB
Name: default
ID: AXXY:B6JJ:6M6W:LXME:VWED:BNEA:4MKU:LDP5:TWKR:CCUH:3SKV:BSBE
Debug mode (server): true
File Descriptors: 28
Goroutines: 46
System Time: 2016-01-16T02:28:33.17774389Z
EventsListeners: 0
Init SHA1:
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
provider=virtualbox
```

**Comment by krishna81m** [comment]
This also helped:
docker-machine stop ..

**Comment by krishna81m** [comment]
Even commits don't work because of the warning which had always been there?
docker commit -a "krishna" -m "cassandra-2.1.12 without ulimit" --pause=true 34d4b228cd55

```
time="2016-01-22T19:39:04.678840891Z" level=debug msg="Calling POST /v1.21/commit"
time="2016-01-22T19:39:04.678878218Z" level=info msg="POST /v1.21/commit?author=krishna+%3E&comment=cassandra-2.1.12+without+ulimit&container=34d4b228cd55&repo=&tag="
time="2016-01-22T19:39:04.678894590Z" level=debug msg="Warning: client and server don't have the same version (client: 1.9.0, server: 1.9.1)"
```

**Comment by yosifkit** [comment]
This looks similar to https://github.com/docker/docker/issues/18180.

**Comment by krishna81m** [comment]
Before giving up on docker for the first time, downgraded to 1.9 and everything works fine:
https://github.com/boot2docker/boot2docker/issues/1099#issuecomment-174440791

**Comment by thaJeztah** [comment]
@krishna81m do you still have this issue on the current boot2docker / docker toolbox? If this is indeed caused by https://github.com/docker/docker/issues/18180, then this should be resolved in the latest version of boot2docker (which has a patched kernel for that issue)

**Comment by thaJeztah** [comment]
I'm going to go ahead and close this, because I think this is fixed in (boot2docker) 1.10, but feel free to continue the discussion
